lib/grape/middleware/error.rb

```diff
@@ -45,6 +45,10 @@ def find_handler(klass)
         handler = options[:rescue_handlers].find(-> { [] }) { |error, _| klass <= error }[1]
         handler ||= options[:base_only_rescue_handlers][klass]
         handler ||= options[:all_rescue_handler]
+        with_option = options[:rescue_options][:with]
+        if with_option.instance_of?(Symbol) && respond_to?(with_option)
```

Comment:
I think we need an exception here, if you do `with: :foobar` and it doesn't exist you want to fail loudly at the time the API is loaded, not at runtime. I am not sure though, it could also happen at runtime. This either way needs a different behavior and tests, we can't just ignore an invalid `with` handler.